skbio/stats/composition.py

```diff
@@ -1701,3 +1711,283 @@ def _check_orthogonality(basis):
+    pvalues will not be reliable. However, the bias is the same across each
+    feature, as a result the ordering of the log-fold changes can still be useful.
+
+    One benefit of using the Dirichlet Multinomial distribution is that the
+    statistical power increases with regards to the abundance magnitude. More counts
+    per sample will shrink the size of the confidence intervals, and can result in
+    lower pvalues.
+
+
+    References
+    ----------
+    .. [1] Fernandes, Andrew D., et al. "Unifying the analysis of
+       high-throughput sequencing datasets: characterizing RNA-seq,
+       16S rRNA gene sequencing and selective growth experiments by
+       compositional data analysis." Microbiome 2 (2014): 1-13.
+    .. [2] Holm, S. "A simple sequentially rejective multiple test procedure".
+       Scandinavian Journal of Statistics (1979), 6.
+
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> from skbio.stats.composition import dirmult_ttest
+    >>> table = pd.DataFrame([[20,  110, 100, 101, 100, 103, 104],
+    ...                       [33,  110, 120, 100, 101, 100, 102],
+    ...                       [12,  110, 100, 110, 100, 50,  90],
+    ...                       [202, 201, 9,  10, 10, 11, 11],
+    ...                       [200, 202, 10, 10, 13, 10, 10],
+    ...                       [203, 201, 14, 10, 10, 13, 12]],
+    ...                      index=['s1', 's2', 's3', 's4', 's5', 's6'],
+    ...                      columns=['b1', 'b2', 'b3', 'b4', 'b5', 'b6',
+    ...                               'b7'])
+    >>> grouping = pd.Series(['treatment', 'treatment', 'treatment',
+    ...                       'placebo', 'placebo', 'placebo'],
+    ...                      index=['s1', 's2', 's3', 's4', 's5', 's6'])
+    >>> lfc_result = dirmult_ttest(table, grouping, 'treatment', 'placebo',
+    ...                            seed=0)
+    >>> lfc_result[["Log2(FC)", "CI(2.5)", "CI(97.5)", "qvalue"]]
+        Log2(FC)   CI(2.5)  CI(97.5)    qvalue
+    b1 -4.991987 -7.884498 -2.293463  0.020131
+    b2 -2.533729 -3.594590 -1.462339  0.007446
+    b3  1.627677 -1.048219  4.750792  0.068310
+    b4  1.707221 -0.467481  4.164998  0.065613
+    b5  1.528243 -1.036910  3.978387  0.068310
+    b6  1.182343 -0.702656  3.556061  0.068310
+    b7  1.480232 -0.601277  4.043888  0.068310
+
+    """
+    rng = get_rng(seed)
+    if not isinstance(table, pd.DataFrame):
+        raise TypeError(
+            "`table` must be a `pd.DataFrame`, " "not %r." % type(table).__name__
+        )
+    if not isinstance(grouping, pd.Series):
+        raise TypeError(
+            "`grouping` must be a `pd.Series`," " not %r." % type(grouping).__name__
+        )
+
+    if np.any(table < 0):
+        raise ValueError("Cannot handle negative values in `table`. ")
+
+    if (grouping.isnull()).any():
+        raise ValueError("Cannot handle missing values in `grouping`.")
+
+    if (table.isnull()).any().any():
```

Comment:
I thought `ndarray.any` without parameter will operate on all dimensions. See [here](https://numpy.org/doc/stable/reference/generated/numpy.any.html). But better double check,

Reply:
this is copied from the `ancom` method. I suggest keeping this as is for consistency - we can always refactor later.